x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/setup_technology.ts

```diff
@@ -94,18 +98,43 @@ export function useSetupTechnology({
     isAgentlessCloudEnabled,
     isAgentlessServerlessEnabled,
   } = useAgentless();
+
   // this is a placeholder for the new agent-BASED policy that will be used when the user switches from agentless to agent-based and back
-  const [newAgentBasedPolicy] = useState<NewAgentPolicy | undefined>({ ...newAgentPolicy });
+  const newAgentBasedPolicy = useRef<NewAgentPolicy>(newAgentPolicy);
   const [selectedSetupTechnology, setSelectedSetupTechnology] = useState<SetupTechnology>(
     SetupTechnology.AGENT_BASED
   );
   const [newAgentlessPolicy, setNewAgentlessPolicy] = useState<AgentPolicy | NewAgentPolicy>(
     generateNewAgentPolicyWithDefaults({
-      name: `Agentless policy ${uuidv4()}`,
       supports_agentless: true,
     })
   );
 
+  useEffect(() => {
```

Comment:
So many `useEffects` are making this hook hard to follow the logic.  We do have tech debt tickets to try to remove the useEffect.

Reply:
This one is meant to trigger changes only when `packagePolicy.name` changes, so I didn't think it would be beneficial to aggregate that logic to the other `useEffect`.

However, I see some room for improvement if we remove all the `useEffects`, but I think a refactor like this will be better handled in this separate [tech debt issue](https://github.com/elastic/kibana/issues/188136) rather than alongside the introduction of a new feature.